ui/src/model-manager/install-models/InatallModelsModal.tsx

```diff
@@ -74,8 +75,8 @@ export default function InatallModelsModal({
   const toast = useToast();
   const [installing, setInstalling] = useState<string[]>([]);
   const [searchQuery, setSearchQuery] = useState(searchQueryProp);
-  const { isOpen, onOpen, onClose } = useDisclosure();
-  const file = useRef<CivitiModelFileVersion>();
+  const { isOpen, onOpen, onClose: onCloseChooseFolderModal } = useDisclosure();
+  const [fileState, setFile, file] = useStateRef<CivitiModelFileVersion>();
```

Comment:
why do we need a file ref and state at the same time? could you help me understand a bit?

Reply:
ref cuz it's used in onClickInstallModel function, state doesn't update inside callback functions.
state cuz it's used in Choose Folders Models to show/hide url input, ref doesn't trigger redraw of DOM.

Reply:
> state doesn't update inside callback functions.
oh really i think normal state will update in onClick() callbacks. only when you put it in a closure like setTimeout() or setInterval(), state will not work, you will need ref. 

or when you didn't add it as the dependency in the useCallback

in other circumstances i think state will update in call backs?